pkg/framework/types.go

```diff
@@ -70,3 +70,9 @@ type BalancePlugin interface {
 	Plugin
 	Balance(ctx context.Context, nodes []*v1.Node) *Status
 }
+
+// PreEvictionFilterPlugin defines an extension point for a general pre eviction filtering operation
+type PreEvictionFilterPlugin interface {
+	Plugin
+	Filter(pod *v1.Pod) bool
```

Comment:
`Filter` is already allocated for filtering pods during listing pods or in the process of determining which pods will get kept untouched. Pre eviction filter needs a new name. E.g. `PreEvictFilter` or similar to signify the filtering is meant to be carried out right before calling the `Evict` method.

I suggest to first create an extension point only for the former case. Introducing the pre eviction filtering extension point in the next PR.

Thus, `s/PreEvictionFilterPlugin/FilterPlugin`.